packages/yoshi-common/src/dev-environment-logger.ts

```diff
@@ -1,44 +1,122 @@
 
-const logSuricateUrls = (appName: string) => {
-  console.log(`  ${chalk.bold('Public:')} ${chalk.magenta(getUrl(appName))}`);
-  console.log();
-  console.log(
-    `Your bundles and other static assets are served from your ${chalk.bold(
-      'dev-server',
-    )}.`,
-  );
-  console.log();
-  console.log(
-    `  ${chalk.bold('Public:')} ${chalk.magenta(getDevServerUrl(appName))}`,
-  );
+const logSuricateUrls = (type: ProcessType, appName: string) => {
+  switch (type) {
+    case 'Server':
+      console.log(
+        `  ${chalk.bold('Public:')} ${chalk.magenta(getUrl(appName))}`,
+      );
+      break;
+    case 'DevServer':
+      console.log(
+        `  ${chalk.bold('Public:')} ${chalk.magenta(getDevServerUrl(appName))}`,
+      );
+      break;
+  }
 };
 
-const logLocalUrls = (serverUrls: Urls, devServerUrls: Urls) => {
-  console.log(
-    `  ${chalk.bold('Local:')}            ${serverUrls.localUrlForTerminal}`,
-  );
-  console.log(
-    `  ${chalk.bold('On Your Network:')}  ${serverUrls.lanUrlForTerminal}`,
-  );
+const logMessageByProcessType: { [type in ProcessType]: string } = {
+  Server: `Your server is starting and should be accessible from your browser.`,
+  Storybook:
+    'Storybook is starting and should be accessible from your browser.',
+  DevServer: `Your bundles and other static assets are served from your ${chalk.bold(
+    'dev-server',
+  )}.`,
+};
 
+const logUrls = ({
+  processType,
+  urls,
+  suricate,
+  appName,
+}: {
+  processType: ProcessType;
+  urls?: Urls;
+  suricate: boolean;
+  appName: string;
+}) => {
   console.log();
-  console.log(
-    `Your bundles and other static assets are served from your ${chalk.bold(
-      'dev-server',
-    )}.`,
-  );
+  console.log(logMessageByProcessType[processType]);
   console.log();
 
-  console.log(
-    `  ${chalk.bold('Local:')}            ${devServerUrls.localUrlForTerminal}`,
-  );
-  console.log(
-    `  ${chalk.bold('On Your Network:')}  ${devServerUrls.lanUrlForTerminal}`,
-  );
+  if (suricate && processType !== 'Storybook') {
+    logSuricateUrls(processType, appName);
+  } else {
+    console.log(
+      `  ${chalk.bold('Local:')}            ${urls?.localUrlForTerminal}`,
+    );
+    console.log(
+      `  ${chalk.bold('On Your Network:')}  ${urls?.lanUrlForTerminal}`,
+    );
+  }
+
+  console.log();
+};
+
+const getProcessName = (type: ProcessType) =>
+  chalk.greenBright(`[${type.toUpperCase()}]`);
+
+const logProcessState = (
+  {
+    processType,
+    suricate,
+    appName,
+  }: { processType: ProcessType; suricate: boolean; appName: string },
+  state: ProcessState,
+) => {
+  switch (state.status) {
+    case 'compiling':
+      console.log(`${getProcessName(processType)}:`, 'Compiling...');
+      break;
+
+    case 'success':
+      logUrls({ processType, suricate, appName, urls: state.urls });
+      break;
+  }
+};
+
+const hasErrorsOrWarnings = (state: State): boolean => {
+  return Object.keys(state).some(stateName => {
```

Comment:
You can use `Object.values` which also removes the line below.
```suggestion
return Object.values(state).some(processState => {
```